### ISSUE TITLE
Accessing IDictionary<_,_>.Count with anonymous type as values gives type constraint mismatch error.

### ISSUE THREAD

**Issue by smoothdeveloper** [dotnet/fsharp#6256]
Accessing `IDictionary<_,_>.Count`  with anonymous type as values gives type constraint mismatch error.

#### Repro steps

In vs2019, edit new .fsx script:

```fsharp
let a = 
    dict
        [
            1,{|a=1;b='2'|}
            2,{|a=2;b='3'|}
        ]

printfn "%i" a.Count
```

#### Expected behavior
Should work the same as with bare tuple or a named record.

#### Actual behavior
> testvs2019.fsx(17,14): error FS0193: Type constraint mismatch. The type 'IDictionary<int,{|a : int ; b : char|}>'    
> is not compatible with type 'ICollection<KeyValuePair<int,{|a : int ; b : char|}>>'    

#### Known workarounds
Use named records or use a function:

```fsharp
    let getCount (d:System.Collections.Generic.IDictionary<_,_>) =
        d.Count
```

#### Related information 
* win10 / vs2019 16.0.0.0 preview 3.0
* may be related to #6214

**Comment by TIHan** [comment]
Can confirm. This seems similar to other issues we've had with constraints and anon records; should be another easy fix once we identified exactly what's happening.

I'll let @cartermp prioritize this.

**Comment by cartermp** [comment]
Let's try to get this in for 16.0 if we can (via servicing). I'll fill out whatever bug we need internally if we can get a fix in promptly.

**Comment by cartermp** [comment]
Given timing I I've moved this out to 16.1.

**Comment by matthid** [comment]
I can confirm as well and hit this just right now.
The workaround is to use 

```fsharp
    let getCount (d:System.Collections.Generic.IDictionary<_,_>) =
        d.Count
```

instead of the `Count` property

**Comment by cartermp** [comment]
Confirmed as resolved for 16.2:

![image](https://user-images.githubusercontent.com/6309070/59707383-60c3d400-91b7-11e9-9311-c270a52e0d27.png)

and in normal projects:

![image](https://user-images.githubusercontent.com/6309070/59707433-7638fe00-91b7-11e9-9c22-c85103f58591.png)
